listenbrainz/webserver/views/playlist_api.py

```diff
@@ -0,0 +1,503 @@
+    :statuscode 400: invalid JSON sent, see error message for details.
+    :statuscode 401: invalid authorization. See error message for details.
+    :resheader Content-Type: *application/json*
+    """
+
+    public = _parse_boolean_arg("public", "true")
+    user = _validate_auth_header()
+
+    data = request.json
+    validate_playlist(data)
+
+    playlist = WritablePlaylist(name=data['playlist']['title'], creator_id=user["id"])
+    playlist.public = public
+
+    if "track" in data["playlist"]:
+        for track in data["playlist"]["track"]:
+            playlist.recordings.append(WritablePlaylistRecording(mbid=UUID(track['identifier'][len(PLAYLIST_TRACK_URI_PREFIX):]),
+                                       added_by_id=user["id"]))
+
+
+    try:
+        playlist = db_playlist.create(playlist)
+    except Exception as e:
+        current_app.logger.error("Error while creating new playlist: {}".format(e))
+        raise APIInternalServerError("Failed to create the playlist. Please try again.")
+
+    return jsonify({'status': 'ok', 'playlist_mbid': playlist.mbid })
+
+
+@playlist_api_bp.route("/<playlist_mbid>", methods=["GET", "OPTIONS"])
+@crossdomain(headers="Authorization, Content-Type")
+@ratelimit()
+def get_playlist(playlist_mbid):
+    """
+    Fetch the given playlist.
+
+    :param playlist_mbid: Optional, The playlist mbid to fetch.
+    :statuscode 200: Yay, you have data!
+    :statuscode 404: Playlist not found
+    :statuscode 401: Invalid authorization. See error message for details.
+    :resheader Content-Type: *application/json*
+    """
+
+    if not is_valid_uuid(playlist_mbid):
+        log_raise_400("Provided playlist ID is invalid.")
+
+    playlist = db_playlist.get_by_mbid(playlist_mbid, True)
+    if playlist is None:
+        raise APINotFound("Cannot find playlist: %s" % playlist_mbid)
+
+    if not playlist.public:
+        user = _validate_auth_header()
+        if playlist.creator_id != user["id"]:
+            raise APINotFound("Cannot find playlist: %s" % playlist_mbid)
+
+    fetch_playlist_recording_metadata(playlist)
+
+    return jsonify(serialize_jspf(playlist))
+
+
+@playlist_api_bp.route("/<playlist_mbid>/item/add/<int:offset>", methods=["POST", "OPTIONS"])
+@playlist_api_bp.route("/<playlist_mbid>/item/add", methods=["POST", "OPTIONS"], defaults={'offset': None})
+@crossdomain(headers="Authorization, Content-Type")
+@ratelimit()
+def add_playlist_item(playlist_mbid, offset):
+    """
+    Append recordings to an existing playlist by posting a playlist with one of more recordings in it.
+    The playlist must be in JSPF format with MusicBrainz extensions, which is defined here:
+    https://musicbrainz.org/doc/jspf .
+
+    If the offset is provided in the URL, then the recordings will be added at that offset,
+    otherwise they will be added at the end of the playlist.
+
+    You may only add MAX_RECORDINGS_PER_ADD recordings in one call to this endpoint.
```

Comment:
We should include it like [here](https://github.com/metabrainz/listenbrainz-server/blob/a233721e/listenbrainz/webserver/views/stats_api.py#L95)